consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/LocalNodeNotProposerTest.java

```diff
@@ -0,0 +1,165 @@
+import static tech.pegasys.pantheon.consensus.ibft.support.MessageReceptionHelpers.assertPeersReceivedNoMessages;
+
+import tech.pegasys.pantheon.consensus.ibft.ConsensusRoundIdentifier;
+import tech.pegasys.pantheon.consensus.ibft.IbftBlockHashing;
+import tech.pegasys.pantheon.consensus.ibft.IbftExtraData;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.CommitPayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.MessageFactory;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.PreparePayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.SignedData;
+import tech.pegasys.pantheon.consensus.ibft.support.RoundSpecificNodeRoles;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContext;
+import tech.pegasys.pantheon.consensus.ibft.support.ValidatorPeer;
+import tech.pegasys.pantheon.crypto.SECP256K1;
+import tech.pegasys.pantheon.crypto.SECP256K1.Signature;
+import tech.pegasys.pantheon.ethereum.core.Block;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class LocalNodeNotProposerTest {
+
+  // This ensures the localNode (i.e. the UUT) will not create the first block
+  private final TestContext context = TestContext.createTestEnvWithUtcClock(4, 0);
+  private final ConsensusRoundIdentifier roundId = new ConsensusRoundIdentifier(1, 0);
+  private final RoundSpecificNodeRoles roles = context.getRoundSpecificRoles(roundId);
+
+  private final MessageFactory localNodeMessageFactory = context.getLocalNodeMessageFactory();
+
+  private final Block blockToPropose = context.createBlockForProposal(0, 15);
+
+  private SignedData<PreparePayload> expectedTxPrepare;
+  private SignedData<CommitPayload> expectedTxCommit;
+
+  @Before
+  public void setup() {
+    expectedTxPrepare =
+        localNodeMessageFactory.createSignedPreparePayload(roundId, blockToPropose.getHash());
+
+    final IbftExtraData extraData = IbftExtraData.decode(blockToPropose.getHeader().getExtraData());
+    final Signature commitSeal =
+        SECP256K1.sign(
+            IbftBlockHashing.calculateDataHashForCommittedSeal(
+                blockToPropose.getHeader(), extraData),
+            context.getLocalNodeParams().getNodeKeys());
+
+    expectedTxCommit =
+        localNodeMessageFactory.createSignedCommitPayload(
+            roundId, blockToPropose.getHash(), commitSeal);
+
+    context.getController().start();
+  }
+
+  @Test
+  public void basicCase() {
+    roles.getProposer().injectProposal(roundId, blockToPropose);
+
+    assertPeersReceivedExactly(roles.getAllPeers(), expectedTxPrepare);
+
+    roles.getNonProposingPeers().get(0).injectPrepare(roundId, blockToPropose.getHash());
+    assertPeersReceivedExactly(roles.getAllPeers(), expectedTxCommit);
+
+    // Ensure the local blockchain has NOT incremented yet.
+    assertThat(context.getBlockchain().getChainHeadBlockNumber()).isEqualTo(0);
```

Comment:
an assert for the block height would be nice, this will probably be used a lot

Reply:
Context now exposes chainheight - good enough?